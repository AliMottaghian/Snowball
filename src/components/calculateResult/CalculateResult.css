.calculateResult .react-clock{
    margin:0 auto
}