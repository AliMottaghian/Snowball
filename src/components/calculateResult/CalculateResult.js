import Clock from "react-clock";
import PropTypes from "prop-types";
import  './CalculateResult.css'

const CalculateResult = ({ timeVal, angleVal }) => {
  return (
    <div className="textCenter calculateResult">
      <Clock value={timeVal} renderSecondHand={false}/>
      <div className="logs">
          <p><strong>Selected time:</strong>{timeVal}</p>
          <p><strong>Angle:</strong>{angleVal}(deg)</p>
      </div>
    </div>
  );
};

CalculateResult.prototype = {
  timeVal: PropTypes.string,
  angleVal: PropTypes.number,
};
export default CalculateResult;
