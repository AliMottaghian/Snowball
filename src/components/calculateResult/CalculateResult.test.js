import CalculateResult from "./CalculateResult";

describe("<CalculateResult />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CalculateResult timeVal="11:20" angleVal="50"/>);
  });
  it("should render CalculateResult component without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render the degree with (deg)", () => {
    const degree = wrapper.find("p").at(1);
    expect(degree.text()).toBe("Angle:50(deg)");
  });
  it("should render two paragraph", () => {
    const paragraph = wrapper.find("p");
    expect(paragraph).toHaveLength(2);
  });
});
