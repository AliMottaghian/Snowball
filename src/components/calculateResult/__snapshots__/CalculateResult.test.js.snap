// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculateResult /> should render CalculateResult component without crashing 1`] = `
<div
  className="textCenter calculateResult"
>
  <Clock
    renderSecondHand={false}
    value="11:20"
  />
  <div
    className="logs"
  >
    <p>
      <strong>
        Selected time:
      </strong>
      11:20
    </p>
    <p>
      <strong>
        Angle:
      </strong>
      50
      (deg)
    </p>
  </div>
</div>
`;
